packages/xod-client-electron/src/arduinoDependencies/actionTypes.js

```diff
@@ -0,0 +1,3 @@
+export const SHOW_UPDATE_ARDUPACKAGES_POPUP = 'SHOW_UPDATE_ARDUPACKAGES_POPUP';
+export const HIDE_UPDATE_ARDUPACKAGES_POPUP = 'HIDE_UPDATE_ARDUPACKAGES_POPUP';
+export const RUN_UPDATE_PROCESS = 'RUN_UPDATE_PROCESS';
```

Comment:
Eh... Flux misuse. Actions are **facts**, not intents. That is:

1. `ARDUPACKAGES_UPDATE_REQUESTED` — it is our decision should any dialog be shown or two, or the upgrade should start right away
2. `POPUP_DISCARDED` — it does not matter which popup, its identifier might be included in the payload in a unified way. The fact is just about clicking the cross icon, “Cancel” button, or ESC key
3. `ARDUPACKAGES_UPGRADE_PROCEED` — the reaction is the same, but semantics is better

Reply:
After googling again for Redux action name conventions, I think the best one to stick to is `<NOUN>_<VERB_INFINITIVE>`. That is:

- `ARDUPACKAGES_UPDATE_REQUEST`
- `POPUP_DISCARD` or `POPUP_CLOSE`
- `ARDUPACKAGES_UPGRADE_PROCEED`

See

- https://decembersoft.com/posts/a-simple-naming-convention-for-action-creators-in-redux-js/
- https://medium.com/@kylpo/redux-best-practices-eef55a20cc72